rskj-core/src/main/java/co/rsk/signing/PlainMessage.java

```diff
@@ -0,0 +1,44 @@
+package co.rsk.signing;
+
+import java.util.Arrays;
+
+/**
+ * Immutable plain byte array message
+ *
+ * @author Ariel Mendelzon
+ */
+public class PlainMessage extends Message {
+    private final byte[] message;
+
+    public PlainMessage(byte[] message) {
+        // Save a copy
+        this.message = copy(message);
+    }
+
+    @Override
+    public byte[] getBytes() {
+        return copy(message);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o == null || this.getClass() != o.getClass()) {
+            return false;
+        }
+
+        return Arrays.equals(this.getBytes(), ((PlainMessage) o).getBytes());
```

Comment:
When you use `getBytes` here you're always copying the array. Use `message` as you did with `hashCode`.

Reply:
:+1: